service/service.go

```diff
@@ -93,13 +104,66 @@ func (s *Service) GetStatus(ctx context.Context, req *pb.UserRequest) (*pb.UserR
 	}, nil
 }
 
-//// CreateUser creates a new user document and inserts it to user DB
-//func (s *Service) CreateUser(ctx context.Context, req *pb.UserRequest) (*pb.UserResponse, error) {
-//	// TODO
-//	log.RequestService("CreateUser")
-//	return &pb.UserResponse{}, nil
-//}
-//
+// CreateUser creates a new user document and inserts it to user DB
+func (s *Service) CreateUser(ctx context.Context, req *pb.UserRequest) (*pb.UserResponse, error) {
+	logger.RequestService("CreateUser")
+
+	// get User object
+	user := req.GetUser()
+	if user == nil {
+		logger.Error(errNilRequestUser.Error())
+		return nil, status.Error(codes.InvalidArgument, errNilRequestUser.Error())
+	}
+
+	// look through mongo user collection to check for email duplication
+	collection := mongoClientWriter.Database(conf.UserDB.Name).Collection(conf.UserDB.Collection)
+	filter := bson.M{"email": user.Email}
+
+	limit := int64(1)
+	docCount, err := collection.CountDocuments(context.TODO(), filter, &options.CountOptions{Limit: &limit})
```

Comment:
When I unit test this part, if the email exists, I get an err saying `...Invalid response from server, value field is not a number`

If email does not exist, no error, it inserts into DB.
I tried removing the Count.Options (b/c it is optional), and I still get the same error.

Reply:
Try https://godoc.org/github.com/mongodb/mongo-go-driver/mongo/options#CountOptions.SetLimit